PencilCase/PencilCase/scripts/application.ts

```diff
@@ -54,18 +54,22 @@ export class Application {
     }
 
     private onDeviceReady = (): void => {
+        let firstPageUrl = window.location.href;
         document.addEventListener('pause', this.onPause, false);
         document.addEventListener('resume', this.onResume, false);
         document.addEventListener('backbutton', function (evt) {
-            //if (cordova.platformId !== 'windows') {
-            //    return;
-            //}
-
-            //if (window.location.href !== firstPageUrl) {
-            //    window.history.back();
-            //} else {
-            //    throw new Error('Exit'); // This will suspend the app
-            //}
+            if (cordova.platformId !== 'android') {
+                return;
+            }
+
+            if (!Application.instance.confirmDialog()) {
+                if (Application.instance.activePage().back)
+                    Application.instance.activePage().back();
+            }
+            else {
+                Application.instance.confirmDialog(null);
```

Comment:
For confirming dialog, other than go to history back, we need to set the dialog instance as null also. Just as the cancel button does automatically.